src/mono/mono/mini/interp/transform.c

```diff
@@ -3437,7 +3440,10 @@ interp_emit_swiftcall_struct_lowering (TransformData *td, MonoMethodSignature *c
 				}
 			} else {
 				// For structs that cannot be lowered, we change the argument to byref type
-				ptype = mono_class_get_byref_type (mono_defaults.typed_reference_class);
+				if (gklass && (gklass->container_class == swift_self_t))
+					ptype = mono_class_get_byref_type (swift_self);
+				else
+					ptype = mono_class_get_byref_type (mono_defaults.int_class);
```

Comment:
As with other comments, don't we just care here that the type is a simple pointer type? Why are we using a byref of a class that doesn't represents the actual contents. Shouldn't we have here, both in case of `SwiftSelf` and `SwiftSelf<T>`, the type being just `m_class_get_byval_arg (mono_defaults.int_class)`, aka a naked pointer passed by value.

Reply:
If `SwiftSelf<T>` can't be lowered, then it should be passed in the same manner as `SwiftSelf`, via the context register. This ensures that the `get_call_info` allocates the correct register, instead of any argument register.